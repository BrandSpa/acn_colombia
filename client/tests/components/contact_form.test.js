import React from "react";
import { shallow, mount, render } from "enzyme";
import ContactForm from "../../components/contact_form";

describe("component ContactForm", () => {
  it("should have 3 inputs and 1 select", () => {
    let wrapper = shallow(<ContactForm />);
    expect(wrapper.find("input").length).toBe(3);
    expect(wrapper.find("select").length).toBe(1);
  });

  it("should change country on componentDidMount", () => {
    let wrapper = mount(<ContactForm country="Germany" />);
    expect(wrapper.state().contact.country).toBe("Germany");
  });

  it("should change contact name", () => {
    let wrapper = shallow(<ContactForm />);
    wrapper
      .find("input")
      .at(0)
      .simulate("change", { e: {}, target: { value: "test" } });
    expect(wrapper.state().contact.name).toEqual("test");
  });
});
