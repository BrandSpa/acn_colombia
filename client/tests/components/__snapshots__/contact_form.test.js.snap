// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component ContactForm should render as expected 1`] = `
Array [
  <form
    className="form-inline contact-form"
    onSubmit={[Function]}
    style={
        Object {
            "textAlign": "center",
          }
    }
>
    <div
        className="input-container"
    >
        <input
            onChange={[Function]}
            placeholder={undefined}
            type="text"
            value=""
        />
        <div
            className="hidden"
        >
             
             
        </div>
    </div>
    <div
        className="input-container"
    >
        <input
            onChange={[Function]}
            placeholder={undefined}
            type="text"
            value=""
        />
        <div
            className="hidden"
        />
    </div>
    <div
        className="input-container"
    >
        <input
            onChange={[Function]}
            placeholder={undefined}
            type="text"
            value=""
        />
        <div
            className="hidden"
        />
    </div>
    <div
        className="input-container"
    >
        <select
            onChange={[Function]}
            value=""
        >
            
        </select>
    </div>
    <button
        onClick={[Function]}
        style={
            Object {
                "marginLeft": "-2px",
              }
        }
    />
</form>,
]
`;
